platform/ios/src/MGLMapView.mm

```diff
@@ -1275,9 +1297,19 @@ - (void)handlePinchGesture:(UIPinchGestureRecognizer *)pinch
         CGFloat newScale = self.scale * pinch.scale;
 
         if (log2(newScale) < _mbglMap->getMinZoom()) return;
-
-        _mbglMap->setScale(newScale, mbgl::ScreenCoordinate { centerPoint.x, centerPoint.y });
-
+        
+        // Calculates the final camera zoom, has no effect within current map camera.
+        MGLMapCamera *toCamera;
+        double zoom = log2(newScale);
```

Comment:
Nit: declare this a few lines further up to avoid saying `log2(newScale)` twice.